modules/internal/ChapelBase.chpl

```diff
@@ -983,6 +983,22 @@ module ChapelBase {
   // sync statements and for joining coforall and cobegin tasks
   //
 
+  inline proc chpl_rt_reset_task_spawn() {
```

Comment:
We don't really need to interface-skip like this, do we?  Rather than calling all the way down into Qthreads from the module code, I'd much rather we add a `chpl_tasks_reset_spawn_order()` to the tasking interface, call that from `chpl_resetTaskSpawn()` below, and then do the usual `CHPL_TASK_IMPL_blah()` trick to only implement it for tasks=qthreads.

Reply:
I can make this change, but can you say more about why you prefer it. I personally never remember to use that approach because I find it really unintuitive for what I have to define and where (and I always have a hard time searching for routines defined in this manner since there's multiple "aliases" for the same name.)


Should this be something like:

chpl-tasks.h
```c
#ifndef CHPL_TASK_IMPL_RESET_SPAWN_ORDER
#define CHPL_TASK_IMPL_RESET_SPAWN_ORDER() 
#endif
static inline
void chpl_task_reset_spawn_order(void) {
  CHPL_TASK_IMPL_RESET_SPAWN_ORDER();
}
```
chpl-tasks-impl.h
```c
#define CHPL_TASK_IMPL_RESET_SPAWN_ORDER() qthread_chpl_reset_spawn_order()
```

or is there a better way to do it?

Reply:
> I can make this change, but can you say more about why you prefer it.

Just software engineering: it simplifies the interface dependencies.  The compiler/module code depends on the tasking layer interface, and then underneath that interface the implementation shim depends on some other library interface (qthreads, pthreads, or massivethreads).  The code in this PR adds yet another dependence, an interface-skip from the compiler/module code directly to the library that underlies one specific tasking implementation.  That makes the code as a whole less understandable, because one has to read more of it in order to see how everything works.  It's better if the compiler/module code depends only on the tasking interface, even if we have to go to a little more effort from time to time to change the latter, to support improvements that only one tasking implementation can support.

I think your example code is exactly right.  Long-term it would make me very happy if we could re-implement the top level of the runtime tasking interface in Chapel in something like `modules/internal/ChapelTasks.chpl` and replace all that `CHPL_TASK_IMPL_*` stuff with param if-stmts, but I'm not holding my breath.

Reply:
Updated